lib/Doctrine/ORM/Query/AST/WhenClause.php

```diff
@@ -13,11 +13,8 @@
  */
 class WhenClause extends Node
 {
-    /**
-     * @param ConditionalExpression $caseConditionExpression
-     * @param mixed                 $thenScalarExpression
-     */
-    public function __construct(public $caseConditionExpression = null, public $thenScalarExpression = null)
+    /** @param ConditionalExpression $caseConditionExpression */
```

Comment:
Apparently, we allow this parameter to be `null`.

```suggestion
    /** @param ConditionalExpression|null $caseConditionExpression */
```

Should be backported to 2.13.x or 2.14.x probably.

Reply:
it's incomplete. 2.13.x is for wrong stuff, 2.14.x for imprecise stuff. I think incomplete here is wrong: it doesn't have to always be a `ConditionalExpression`

Reply:
Looking at #10224: The `$caseConditionExpression` parameter was not optional on 2.x. Why is it here? Let's remove the default and document it as not nullable instead?

Reply:
:thinking: indeed. I think that must be a choice rector makes between the doc of the property, and the doc of the parameter. I'll amend #10224 after confirming it's safe to do so

Reply:
Using `object` as a type hint passes the test suite (and `ConditionalExpression` does not, of course lol)